reactive/src/main/scala/outwatch/reactive/Sink.scala

```diff
@@ -0,0 +1,17 @@
+package outwatch.reactive
+
+trait Sink[-F[_]] {
+  def onNext[A](sink: F[A])(value: A): Unit
+  def onError[A](sink: F[A])(error: Throwable): Unit
```

Comment:
@cornerman In order to be functionally pure, the `onNext` and `onError` in `Sink` need to return something other than `Unit`.

Reply:
You are right, the same holds for the `subscribe` method on the `Source`. It should return an effect type.